subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/impl/DirectorySnapshotter.java

```diff
@@ -274,9 +275,9 @@ private CompleteFileSystemLocationSnapshot snapshotFile(Path absoluteFilePath, S
             String internedAbsoluteFilePath = intern(remapAbsolutePath(absoluteFilePath));
             if (attrs.isRegularFile()) {
                 try {
-                    HashCode hash = hasher.hash(absoluteFilePath.toFile(), attrs.size(), attrs.lastModifiedTime().toMillis());
-                    FileMetadata metadata = FileMetadata.from(attrs);
-                    return new RegularFileSnapshot(internedAbsoluteFilePath, internedName, hash, metadata, accessType);
+                    FileMetadataSnapshot metadata = DefaultFileMetadataSnapshot.file(attrs.lastModifiedTime().toMillis(), attrs.size(), accessType);
+                    HashCode hash = hasher.hash(absoluteFilePath.toFile(), metadata.getLength(), metadata.getLastModified());
```

Comment:
Shall we extract these things to local variables instead?

Reply:
Extracted.